components/builder-scheduler/src/data_store.rs

```diff
@@ -586,4 +653,134 @@ impl DataStore {
 
         Ok(groups)
     }
+
+    pub fn create_job_status(&self, msg: &JobStatus) -> Result<()> {
+        let conn = self.pool.get_shard(0)?;
+        let job = msg.get_job();
+
+        let job_id = job.get_id() as i64;
+        let owner_id = job.get_owner_id() as i64;
+
+        let job_state = match job.get_state() {
+            JobState::Dispatched => "Dispatched",
+            JobState::Pending => "Pending",
+            JobState::Processing => "Processing",
+            JobState::Complete => "Complete",
+            JobState::Rejected => "Rejected",
+            JobState::Failed => "Failed",
```

Comment:
Should we implement `FromStr` (so we get can use `parse`) and `Display` for `JobState` instead, to handle the conversion to and from a String?